### ISSUE TITLE
Adds same tabs multiple times

### ISSUE THREAD

**Issue by ealouie** [dougdellolio/opensource#10]
@dougdellolio @kakawangza 

Testing out the Doug's latest commit (f7c41a8bc56a3e1a5d6fe02586c27654ab19587a), I found that when I repeatedly clicked on the extension it would add the same tabs again (basically duplicate the first correct list, and append it to the bottom).

Steps to duplicate:
Open Chromium, click on extension, close extension, click on extension again (this time it should be correct with only one set of tabs in the history), close extension again, open extension again (this is when there should be an identical list appended to the bottom), and then it keeps repeating when you do that.

**Comment by ealouie** [comment]
I think it would be more apparent what was going on if the state of the tabs was updated, and that was posted as opposed to it repeatedly being appended to the bottom (especially if you have exactly the same tabs open)? Maybe that is just me though. I'll look into a way to do that unless you guys like it the way it is.

**Comment by dougdellolio** [comment]
The reason it is being added over and over is because I call the function addtohistory() everytime the popup is pressed. What we need to do is read when chrome is closed and only call the function once just before the exit. That was something I couldn't quite figure out how to do. I can look at it again soon

**Comment by ealouie** [comment]
Yeah, I'm looking into it now. I'll let you guys know what I find.

**Comment by ealouie** [comment]
@dougdellolio Just found the Chrome sessions API (https://developer.chrome.com/extensions/sessions). Could this make everything a lot simpler?
